lib/widgets/app.dart

```diff
@@ -139,6 +139,50 @@ class ZulipApp extends StatefulWidget {
 }
 
 class _ZulipAppState extends State<ZulipApp> with WidgetsBindingObserver {
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addObserver(this);
+  }
+
+  @override
+  void dispose() {
+    WidgetsBinding.instance.removeObserver(this);
+    super.dispose();
+  }
+
+  List<Route<dynamic>> _handleGenerateInitialRoutes(
+    BuildContext context,
+    String initialRoute,
+  ) {
+    final globalStore = GlobalStoreWidget.of(context);
+    final initialRouteUrl = Uri.tryParse(initialRoute);
+    if (initialRouteUrl case Uri(scheme: 'zulip', host: 'notification')) {
+      final route = NotificationDisplayManager.routeForNotification(
+        globalStore: globalStore,
+        url: initialRouteUrl);
+
+      if (route != null) {
+        return [
+          HomePage.buildRoute(accountId: route.accountId),
+          route,
+        ];
+      } else {
+        // The account didn't match any existing accounts,
+        // fallthrough to show default the route below.
+      }
```

Comment:
nit:
```suggestion
        // The account didn't match any existing accounts,
        // fallthrough to show the default route below.
      }
```

?

Reply:
ah and another nit: "fall through" is the verb, while "fallthrough" is a noun

(similar to "clean up" / "cleanup" and so on, as in https://github.com/zulip/zulip-flutter/pull/1240#discussion_r1936369343 )